libtransmission/session.h

```diff
@@ -243,6 +243,7 @@ struct tr_session
         [[nodiscard]] std::optional<std::string> bind_address_V6() const override;
         [[nodiscard]] std::optional<std::string_view> userAgent() const override;
         [[nodiscard]] size_t clamp(int torrent_id, size_t byte_count) const override;
+        [[nodiscard]] std::optional<std::string_view> proxyUrl() const override;
```

Comment:
The other suggestions are optional, but this is important.

`std::string_view` aren't guaranteed to be null-terminated, so it's not appropriate to use it here.

```suggestion
        [[nodiscard]] std::optional<std::string> proxyUrl() const override;
```

P.S. I know `userAgent()` also uses `std::string_view`, and I think that needs to be changed too. But that's outside of the scope of this PR.